pkg/controller/dnsendpoint/controller.go

```diff
@@ -193,37 +198,43 @@ func (d *controller) processItem(key string) error {
 
 	if !exists {
 		//delete corresponding DNSEndpoint object
-		return d.client.MulticlusterdnsV1alpha1().DNSEndpoints(namespace).Delete(name, &metav1.DeleteOptions{})
-	}
-
-	dnsEndpoints, err := d.getEndpoints(obj)
-	if err != nil {
-		return err
-	}
+		err = d.client.Get(context.TODO(), objectKey, dnsEndpointObject)
```

Comment:
It isn't necessary to retrieve an object to delete it.  Setting the name and namespace in the object meta of an object of the target type (e.g. DNSEndpoint in this case) is sufficient.

Reference: [Examples in the controller-runtime repo](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/example_test.go#L177)